contributed_definitions/nyaml/NXxps.yaml

```diff
@@ -0,0 +1,474 @@
+category: application
+doc: |
+  This is the application definition for X-ray photoelectron spectroscopy.
+type: group
+NXxps(NXmpes):
+  (NXentry):
+    definition(NX_CHAR):
+      enumeration: [NXxps]
+    method:
+      doc: |
+        A name of the experimental method according to `Clause 11`_ of
+        the `ISO 18115-1:2023`_ specification.
+        
+        Examples for XPS-related experiments include:
+            * X-ray photoelectron spectroscopy (XPS)
+            * angle-resolved X-ray photoelectron spectroscopy (ARXPS)
+            * ultraviolet photoelectron spectroscopy (UPS)
+            * hard X-ray photoemission spectroscopy (HAXPES)
+            * near ambient pressure X-ray photoelectron spectroscopy (NAPXPS)
+            * electron spectroscopy for chemical analysis (ESCA)
+        
+        .. _ISO 18115-1:2023: https://www.iso.org/standard/74811.html
+        .. _Clause 11: https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:sec:11
+    geometries(NXcoordinate_system_set):
+      xps_coordinate_system(NXcoordinate_system):
+        exists: optional
+        doc: |
+          In traditional surface science, the coordinate system is defined such that the positive z-axis
+          points along the normal of the sample stage, and the x- and y-axes lie in the plane of the sample stage.
+          However, in NeXus, a coordinate system that is the same as `McStas`_  is used. `xps_coordinate_system`
+          gives the user the opportunity to work in the traditional base coordinate system.
+          
+          .. _McStas: http://mcstas.org/
+          
+          .. image:: xps/xps_cs.png
+              :width: 40%
+        origin:
+          enumeration: [sample stage]
+        handedness:
+          enumeration: [left_handed]
+        z_direction:
+          enumeration: [sample stage normal]
+        depends_on(NX_CHAR):
+          doc: |
+            Link to transformations defining the XPS base coordinate system,
+            which is defined such that the positive z-axis points along the sample stage normal, and the
+            x- and y-axes lie in the plane of the sample stage.
+        (NXtransformations):
+          doc: |
+            Set of transformations, describing the orientation of the XPS coordinate system
+            with respect to the beam coordinate system (.).
+    (NXinstrument):
+      doc:
+      - |
+        Description of the XPS spectrometer and its individual parts.
+      - |
+        xref:
+          spec: ISO 18115-1:2023
+          term: 12.58
+          url: https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:term:12.58
+      sourceTYPE(NXsource):
+        power(NX_FLOAT):
+          unit: NX_POWER
+          exists: recommended
+        transformations(NXtransformations):
+          exists: recommended
+          source_polar_angle_of_incidence(NX_NUMBER):
+            unit: NX_ANGLE
+            doc: |
+              Incidence angle
```

Comment:
The source is the piece that generates the beam, e.g. a synchrotron or a x-ray tube or so. An incidence angle I would rather associate with the corresponding beam (?)

Reply:
In the ISO standard, the angles are defined wrt the source. However, I agree that these are actually angles associated with the beam and they are linked through the `associated_source` field, so I just moved these angles to the beam.